
syntax = "proto3";

package common;

// This message contains the core data extracted from an invoice document.
// It is used by multiple services in the workflow.
message ExtractedInvoiceData {
    string invoice_no = 1;
    string vendor_gstin = 2;
    float total_amount = 3;
    repeated ItemDetail item_details = 4;
    float confidence_score = 5;
}

// Represents a single line item within an invoice.
message ItemDetail {
    string item_description = 1;
    float unit_price = 2;
    float quantity = 3;
}
